internal/webserver/server.go

```diff
@@ -173,9 +180,95 @@ func (webserver *WebServer) versionHandler(writer http.ResponseWriter, _ *http.R
 	return
 }
 
+// swagger:operation POST /secrets Secret
+//
+// Secret
+//
+// Posts secret(s) to the SDK that the SDK will add to the secret client.
+//
+// ---
+//
+// requestBody:
+// content:
+//   application/json:
+// 	schema:
+// 	  $ref: '#/definitions/SecretData'
+// 	example:
+//  {
+//   "path" : "MyPath"
+//   "secrets":[
+//       {
+//         "key" : "MySecretKey",
+//         "value" : "MySecretValue"
+//       }
+//    ]
+//  }
+//
+// produces:
+//
+// Schemes:
+//  - http
+//
+// Responses:
+//  '201':
+//    description: Posted secret(s) to the SDK
+//    schema:
+//      "$ref": "#/definitions/SecretData"
+//  '400':
+//    description: Bad request could not decode JSON payload
+//    schema:
+//      "$ref": "#/definitions/SecretData"
+//  '500':
+//    description: Internal Server Error
+//    schema:
+//      "$ref": "#/definitions/SecretData"
+//
+func (webserver *WebServer) secretHandler(writer http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(req.Body)
+	var secretData SecretData
+
+	err := decoder.Decode(&secretData)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to parse json payload: %v", err)
+		webserver.writeResponse(writer, msg, http.StatusBadRequest)
+		return
+	}
+
+	if err = secretData.validateSecretData(); err != nil {
+		webserver.writeResponse(writer, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var secretsKV = make(map[string]string)
+	for _, secret := range secretData.Secrets {
+		secretsKV[secret.Key] = secret.Value
+	}
+
+	if err := webserver.secretProvider.StoreSecrets(secretData.Path, secretsKV); err != nil {
+		msg := fmt.Sprintf("Storing secret failed: %v", err)
+		webserver.writeResponse(writer, msg, http.StatusInternalServerError)
+		return
+	}
+
+	writer.WriteHeader(http.StatusCreated)
+	return
+}
+
+func (webserver *WebServer) writeResponse(writer http.ResponseWriter, msg string, statusCode int) {
+	msg = "Missing required field 'Secrets'"
+	webserver.LoggingClient.Error(msg)
+
+	writer.WriteHeader(http.StatusBadRequest)
```

Comment:
statusCode isn't being used. Replace the status here with statusCode?

Reply:
Ah good catch thank you!